### DISCUSSION TITLE
writing characteristic failed

### DISCUSSION THREAD

**Discussion by dmsuh** [capacitor-community/bluetooth-le#538] [Q&A]
![image](https://github.com/capacitor-community/bluetooth-le/assets/90576894/8aee0b5f-be60-4488-9d85-84aba10941f3)
I'm recording according to the rules. But nothing works out.
![image](https://github.com/capacitor-community/bluetooth-le/assets/90576894/2a3a92d8-e511-4dcd-bb3d-26092005eb38)
This entry has a code to restore to factory settings
![image](https://github.com/capacitor-community/bluetooth-le/assets/90576894/197f9bd6-29f2-4500-9bb2-cf8c43ce565c)
The test program outputs the result perfectly

**Comment by dmsuh** [comment]
additional screen from logcat
![image](https://github.com/capacitor-community/bluetooth-le/assets/90576894/0d253c44-08e4-4215-b1dc-b0f2ec941621)

**Reply by pwespi** [reply]
Can you write to the characteristic with an app like nRF connect mobile?

**Reply by dmsuh** [reply]
yes its app write characteristic

**Reply by dmsuh** [reply]
But why its working in my code?

**Reply by dmsuh** [reply]
screenshot attached ,in the program at right, check the received notification from device
![image](https://github.com/capacitor-community/bluetooth-le/assets/90576894/e91c8a30-3410-4f86-a5ca-beed3e91589d)

**Reply by dmsuh** [reply]
my problem is here 
https://stackoverflow.com/a/51095280

**Reply by dmsuh** [reply]
The recording goes through but the error goes through, because callback dont register.
I'll try recording without a callback.
Use it https://github.com/capacitor-community/bluetooth-le#writewithoutresponse

**Reply by pwespi** [reply]
> my problem is here https://stackoverflow.com/a/51095280

I don't understand how the connectGatt call is related.
